testing/e2e/suites/defaultBackendService.go

```diff
@@ -207,8 +270,13 @@ var defaultBackendClientServerTest = func(ctx context.Context, config *rest.Conf
 				}
 			}
 
-			testingResources = manifests.DefaultBackendClientAndServer(zoneNamespace, zoneName, zone.GetNameserver(), zoneKVUri, ingressClassName, zoneHost, tlsHost)
-			nic.Spec.DefaultBackendService = &v1alpha1.NICNamespacedName{"default-" + zoneName + "-service", zoneNamespace}
+			if nic.Spec.CustomHTTPErrors != nil && len(nic.Spec.CustomHTTPErrors) > 1 {
+				testingResources = manifests.CustomErrorsClientAndServer(zoneNamespace, zoneName, zone.GetNameserver(), zoneKVUri, zoneHost, tlsHost, ingressClassName, serviceName)
```

Comment:
wonder if it would be better to have the manifests package take in the nic you define and determine which infra to provision based on, in this case, the custom errors we want to test for

Reply:
I think that's a good idea. I can work on that and make a new PR for that refactor!

Reply:
sounds good!